bridge-ffi/src/lib.rs

```diff
@@ -0,0 +1,650 @@
+#![allow(non_camel_case_types)]
+
+use prost::Message;
+use std::str::FromStr;
+use temporal_sdk_core_protos::coresdk::bridge;
+
+extern crate libc;
+
+/// A set of bytes owned by Core. This must always be passed to
+/// tmprl_bytes_free when no longer in use.
+#[repr(C)]
+pub struct tmprl_bytes_t {
+    bytes: *const u8,
+    len: libc::size_t,
+    cap: libc::size_t,
+    disable_free: bool,
+}
+
+impl tmprl_bytes_t {
+    fn from_vec(vec: Vec<u8>) -> tmprl_bytes_t {
+        // Mimics Vec::into_raw_parts that's only available in nightly
+        let mut vec = std::mem::ManuallyDrop::new(vec);
+        tmprl_bytes_t {
+            bytes: vec.as_mut_ptr(),
+            len: vec.len(),
+            cap: vec.capacity(),
+            disable_free: false,
+        }
+    }
+
+    fn from_vec_disable_free(vec: Vec<u8>) -> tmprl_bytes_t {
+        let mut b = tmprl_bytes_t::from_vec(vec);
+        b.disable_free = true;
+        b
+    }
+
+    fn into_raw(self) -> *mut tmprl_bytes_t {
+        Box::into_raw(Box::new(self))
+    }
+}
+
+// We only impl this because they are required by lazy_static and raw pointers
+// are not usually safe for send
+unsafe impl Send for tmprl_bytes_t {}
+unsafe impl Sync for tmprl_bytes_t {}
+
+impl Drop for tmprl_bytes_t {
+    fn drop(&mut self) {
+        // In cases where freeing is disabled (or technically some other
+        // drop-but-not-freed situation though we don't expect any), the bytes
+        // remain non-null so we re-own them here
+        if !self.bytes.is_null() {
+            unsafe { Vec::from_raw_parts(self.bytes as *mut u8, self.len, self.cap) };
+        }
+    }
+}
```

Comment:
Nah, when we do `Box::from_raw`, you disable dropping because you remove ownership. The only reason I added this was for static ones, and they'd only be dropped on entire library unload. It's just a memory safety mechanism for those cases since `tmprl_bytes_free` is disabled for them.

Reply:
K, shouldn't this check that field then?

Reply:
That is the purpose of one part of the comment there: 

> or technically some other drop-but-not-freed situation though we don't expect any

So I check whether the pointer is null instead of whether freeing was disabled (during explicit free, we set this pointer as null for this reason)